pubtools/exodus/hooks/pulp.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import os
+import sys
+
+import backoff
+import requests
+from pubtools.pluggy import hookimpl, pm  # pylint: disable=wrong-import-order
+from six.moves.urllib.parse import urljoin
+
+LOG = logging.getLogger("pubtools-exodus")
+
+
+class ExodusPulpHanlder:
+    def __init__(self):
+        self._gw_url = os.getenv("EXODUS_GW_URL")
+        self._gw_env = os.getenv("EXODUS_GW_ENV")
+        self._gw_crt = os.getenv("EXODUS_GW_CERT")
+        self._gw_key = os.getenv("EXODUS_GW_KEY")
+
+        self._publish = None
+        self._check_cert()
+
+    def _new_session(self):
+        out = requests.Session()
+        out.cert = (self._gw_crt, self._gw_key)
+        return out
+
+    def _do_request(self, **kwargs):
```

Comment:
Could we please have some retry logic here as well, to retry individual requests?